File: Python/Librerias.py
# En los ejercicios anteriores hemos visto dos bibliotecas de Python
import math  # para operaciones matemáticas complejas
import sys   # que guarda las funciones del sistema, como terminar procesos, entre otras.
# Pero estas no son las únicas que existen.
# Entre las bibliotecas de Python se destacan dos tipos: las estándar, que vienen con el lenguaje y son fáciles de implementar,
# y las que debemos instalar nosotros adicionamente.
# Aquí solo repasaremos las más útiles, pero si desea más información puede visitar esta página:
# https://docs.python.org/es/3/library/index.html

# Bibliotecas estándar de Python
import os    # para interactuar con el sistema operativo, como leer o escribir en el sistema de archivos
import time  # para medir el tiempo, realizar pausas y trabajar con fechas
import random  # para generar números aleatorios
import urllib.request  # para trabajar con URLs y realizar solicitudes web
print(sys.version)  # muestra la versión de Python que estamos utilizando

# Ejemplo básico de uso de una biblioteca
# Obtener el directorio actual usado el import os
directorio_actual = os.getcwd()
print("Directorio actual:", directorio_actual)
# Crear un nuevo directorio
nuevo_directorio = "nuevo_directorio"
os.mkdir(nuevo_directorio)
print("Se creó un nuevo directorio:", nuevo_directorio)
# Eliminar el directorio creado
os.rmdir(nuevo_directorio)
print("Se eliminó el directorio:", nuevo_directorio)
#import time
# Obtener la hora actual
hora_actual = time.strftime("%H:%M:%S")
print("Hora actual:", hora_actual)
# Pausa de 2 segundos
print("Esperando 2 segundos...")
time.sleep(2)
print("Pausa completada.")
# Generar un número aleatorio entre 1 y 10
numero_aleatorio = random.randint(1, 10)
print("Número aleatorio:", numero_aleatorio)
#import urllib.request 
# Descargar una página web
url = "https://www.ejemplo.com"
respuesta = urllib.request.urlopen(url)
contenido = respuesta.read().decode('utf-8')

File: C/Librerias.c
// Liberias de lenguaje c 
#include <stdio.h> //Esta librería proporciona funciones estándar de entrada y salida
#include <time.h>
//Se utiliza para realizar la función printf, que permite mostrar mensajes.
/*
* En C y C++, el sufijo ".h" en los nombres de archivo de encabezado (header files) 
* se utiliza para indicar que se trata de un archivo de cabecera (header file). 
* Estos archivos suelen contener declaraciones de funciones, macros, constantes y 
* otras declaraciones que se utilizan en varios archivos fuente del programa.
*/
// tambien podemos incluir una libreria con ""
#include "string.h" //Proporciona funciones para gestión de memoria dinámica.
int main() {
    printf("Hola, gente!\n");
    int numero;
    printf("Ingrese un número: ");
    scanf("%d", &numero);
    printf("Usted ingresó: %d\n", numero);
    srand(time(NULL));
    // Generar y mostrar tres números aleatorios entre 1 y 10
    for (int i = 0; i < 3; ++i) {
        int numeroAleatorio = rand() % 10 + 1;
        printf("Número aleatorio %d: %d\n", i + 1, numeroAleatorio);
    }
    return 0;
}